File: app/src/main/java/com/arny/flightlogbook/data/db/FlightTypeDAO.java
package com.arny.flightlogbook.data.db;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;
import com.arny.flightlogbook.data.models.FlightType;

import java.util.List;

@Dao
public interface FlightTypeDAO {
	@Query("SELECT * FROM flight_type")
	List<FlightType> getFlightTypes();

	@Query("SELECT * FROM flight_type WHERE _id=:id")
	FlightType getFlightType(long id);

	@Query("DELETE FROM flight_type WHERE _id=:id")
	int delete(long id);

	@Query("DELETE FROM flight_type")
	int delete();

	@Insert(onConflict = OnConflictStrategy.REPLACE)
	long insert(FlightType flightType);
}

File: app/src/main/java/com/arny/flightlogbook/data/db/daos/FlightTypeDAO.kt
package com.arny.flightlogbook.data.db.daos

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.arny.flightlogbook.data.db.intity.flights.FlightEntity
import com.arny.flightlogbook.data.models.flights.FlightTypeEntity

@Dao
interface FlightTypeDAO {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertReplace(items: Collection<FlightTypeEntity>): Array<Long>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(item: FlightTypeEntity): Long

    @Query("SELECT * FROM flight_type")
    fun queryTypes(): List<FlightTypeEntity>

    @Query("SELECT * FROM flight_type WHERE _id=:id")
    fun queryFlightType(id: Long?): FlightTypeEntity?

    @Query("UPDATE flight_type SET title=:title WHERE _id=:id ")
    fun setTitle(id: Long?, title: String?): Int

    @Query("DELETE FROM flight_type WHERE _id=:id")
    fun delete(id: Long?): Int

    @Query("DELETE FROM flight_type")
    fun delete(): Int
}
